troposphere/static/js/components/modals/provider_machine/ProviderMachineEditModal.react.js

```diff
@@ -0,0 +1,250 @@
+/** @jsx React.DOM */
+
+define(
+  [
+    'react',
+    'stores',
+    'components/mixins/BootstrapModalMixin.react',
+    'components/common/image_request/ImageVisibility.react',
+    'components/applications/detail/availability/AvailabilityView.react',
+    'components/applications/detail/description/EditDescriptionView.react',
+    'components/modals/provider_machine/Application.react'
+  ],
+  function (React, stores, BootstrapModalMixin, Visibility, AvailabilityView, EditDescriptionView, Application) {
+
+    return React.createClass({
+      mixins: [BootstrapModalMixin],
+
+      propTypes: {
+        machine: React.PropTypes.instanceOf(Backbone.Model).isRequired,
+        application: React.PropTypes.instanceOf(Backbone.Model).isRequired
+      },
+
+      getInitialState: function () {
+        return this.getState();
+      },
+      updateState: function () {
+        if (this.isMounted()) this.setState(this.getState());
+      },
+      getState: function() {
```

Comment:
New code (like this modal) should probably move AWAY from the 'getState' approach and move the 'heavy lifting' code into render (if it can change -- as suggested by @prosif) or getInitialState (If it is set once and available on 'initialization')..